nixos/modules/services/monitoring/scrutiny-collector.nix

```diff
@@ -0,0 +1,123 @@
+
+  prettyJSON = conf:
+    pkgs.runCommand "loki-config.json" { } ''
+      echo '${builtins.toJSON conf}' | ${pkgs.jq}/bin/jq 'del(._module)' > $out
+    '';
+
+in
+{
+  options.services.scrutiny-collector = {
+    enable = mkEnableOption "scrutiny-collector";
+
+    package = mkPackageOption pkgs "scrutiny-collector" { };
+
+    api-endpoint = mkOption {
+      type = types.str;
+      default = "http://localhost:8080";
+      description = mdDoc ''
+        The api server endpoint
+      '';
+    };
+
+    log-file = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      description = mdDoc ''
+        Path to file for logging. Leave empty to use STDOUT
+      '';
+    };
+
+    debug = mkEnableOption "Enable debug logging";
+
+    host-id = mkOption {
+      type = types.str;
+      default = config.networking.hostName;
+      defaultText = lib.literalExpression "config.networking.hostName";
+      description = mdDoc ''
+        Host identifier/label, used for grouping devices
+      '';
+    };
+
+    calendar = mkOption {
+      type = types.str;
+      default = "*-*-* *:00,15,30,45:00";
+      description = mdDoc ''
+        Configured when to run the service systemd unit (DayOfWeek Year-Month-Day Hour:Minute:Second).
+      '';
+    };
+
+    user = mkOption {
+      type = types.str;
+      default = "scrutiny-collector";
+      description = mdDoc ''
+        User under which the scrutiny collector service runs.
+      '';
+    };
+
+    group = mkOption {
+      type = types.str;
+      default = "scrutiny-collector";
+      description = mdDoc ''
+        Group under which the scrutiny collector service runs.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    environment.systemPackages = [ cfg.package ];
+
+    users.groups.${cfg.group} = { };
+    users.users.${cfg.user} = {
+      description = "Scrutiny Collector Service User";
+      group = cfg.group;
+      createHome = false;
+      isSystemUser = true;
+    };
+
+    systemd = {
+      timers.scrutiny-collector = {
+        description = "Scrutiny Collector";
+        wantedBy = [ "timers.target" ];
+
+        timerConfig = {
+          OnCalendar = cfg.calendar;
+          AccuracySec = "5m";
+          Unit = "scrutiny-collector.service";
+        };
+      };
+
+      services.scrutiny-collector = {
+        description = "Scrutiny Collector";
+
+        serviceConfig =
+          let
+            cmd = "${cfg.package}/bin/collector-metrics run";
+            args = lib.pipe [
+              (if cfg.api-endpoint != null then "--api-endpoint ${cfg.api-endpoint}" else null)
+              (if cfg.log-file != null then "--log-file ${cfg.log-file}" else null)
+              (if cfg.debug then "--debug" else null)
+              (if cfg.host-id != null then "--host-id ${cfg.host-id}" else null)
+            ] [
+              (lib.filter (arg: arg != null))
+              (lib.concatStringsSep " ")
+            ];
```

Comment:
```suggestion
            args = lib.cli.toGNUCommandLineShell { } {
              inherit (cfg) api-endpoint log-file debug host-id;
            };
```

Reply:
Oh, that's pretty neat. How would the debug flag be handled? It shouldn't be present if cfg.debug is false.